crates/autopilot/src/run_loop.rs

```diff
@@ -510,11 +513,13 @@ impl RunLoop {
 
         // Shuffle so that sorting randomly splits ties.
         solutions.shuffle(&mut rand::thread_rng());
-        solutions.sort_unstable_by_key(|participant| participant.solution.score().get().0);
+        solutions.sort_unstable_by_key(|participant| {
+            std::cmp::Reverse(participant.solution.score().get().0)
+        });
 
         // Make sure the winning solution is fair.
-        while !Self::is_solution_fair(solutions.last(), &solutions, auction) {
-            let unfair_solution = solutions.pop().expect("must exist");
+        while !Self::is_solution_fair(solutions.first(), &solutions, auction) {
+            let unfair_solution = solutions.remove(0);
```

Comment:
Due to changing the sorting a few lines above we now have this inefficient `.remove(0)`. Why is there a need to reverse the order?

Reply:
Commented on reasons below 👇 

> Due to changing the sorting a few lines above we now have this inefficient .remove(0)

I'd say this is just theoretical in this case, TBH, I don't think we would see any differences in practice. But if you are worried about it, I would still prefer to collect solutions into `VecDeque` or something, and keep the ordering.

Reply:
Then let's use a `VecDeque`. Calling `.remove(0)` on a vector curls my toenails. 😅 (even though I also expect this to not be a significant issue).

Reply:
Switched to `VecDeque `